values.go

```diff
@@ -1419,7 +1424,11 @@ func (v Fix64) MeteredType(common.MemoryGauge) Type {
 }
 
 func (v Fix64) ToGoValue() any {
-	return int64(v)
+	value, err := strconv.ParseFloat(v.String(), 64)
+	if err != nil {
+		panic(err)
+	}
+	return value
```

Comment:
Fixed point numbers are not floating point numbers. This potentially a lossy conversion, as some fixed-point values cannot be represented as floating point numbers.

Reply:
Interesting, what situations would that occur in?